### DISCUSSION TITLE
Poetry Extras

### DISCUSSION THREAD

**Discussion by AleksandrLiadov** [python-poetry/poetry#6426] [General] [answered]
Guys, could you help me, please.
I have a core package where I specify an extra dep (ex. google_dev). (so poetry **install -E google_dev**)
So when I install my core package in another project, I dont have optional packages from google_dev (because I didn't install it).
Imagine, In this project I want to install a package which will conflict with extra (optional) package from core package.
I'm agree if extra deps were installed, so the conflict package cannot be installed, however if extra deps were not installed I should be able to install this conflict package, but I cannot 😦

**Comment by neersighted** [comment, marked as answer]
Currently every possible combination of packages specified in `pyproject.toml` must be mutually installable. There is currently no support for mutually exclusive deps, but early discussion of this is taking place at #6419.
